packages/mobile/src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,349 @@
+import Button from '@celo/react-components/components/Button'
+import Card from '@celo/react-components/components/Card'
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import { Shadow } from '@celo/react-components/styles/styles'
+import variables from '@celo/react-components/styles/variables'
+import React, { useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import {
+  ActivityIndicator,
+  Image,
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+} from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { dappsListApiUrlSelector } from 'src/app/selectors'
+import BackButton from 'src/components/BackButton'
+import BottomSheet from 'src/components/BottomSheet'
+import Dialog from 'src/components/Dialog'
+import CustomHeader from 'src/components/header/CustomHeader'
+import DappsExplorerLogo from 'src/icons/DappsExplorerLogo'
+import Help from 'src/icons/navigator/Help'
+import QuitIcon from 'src/icons/QuitIcon'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import useSelector from 'src/redux/useSelector'
+import { navigateToURI } from 'src/utils/linking'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'DAppExplorerScreen'
+
+interface CategoryWithItems {
+  id: string
+  name: string
+  fontColor: string
+  backgroundColor: string
+  items: DappItem[]
+}
+
+interface DappItem {
+  id: string
+  iconUrl: string
+  name: string
+  description: string
+  dappUrl: string
+}
+
+export function DAppsExplorerScreen() {
+  const { t } = useTranslation()
+  const dappsListUrl = useSelector(dappsListApiUrlSelector)
+  const [categoriesWithItems, setcategoriesWithItems] = useState<CategoryWithItems[]>([])
+  const [isHelpDialogVisible, setHelpDialogVisible] = useState(false)
+  const [isBottomSheetVisible, setBottomSheetVisible] = useState(false)
+  const [dappSelected, setDappSelected] = useState<DappItem>()
+
+  if (!dappsListUrl) {
+    return (
```

Comment:
i think this guy should be moved further down the component, just before the last `return` because if react will complain about mismatched number hooks between re-renders if the code does return early here :) (alternatively `useAsync` could be moved before this return, but it's easier to accidentally add another hook after this return later on by accident) (https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

Reply:
You are right, already changed :)